### PR TITLE
tests: get rid of skip_ui

### LINKED ISSUE #1409: Re-enable skipped UI tests (or document why they are skipped)
Currently we skip lots of UI tests in the device_tests suite.

Let's re-enable them where possible or document (via a comment) why the test needs to have disabled UI test.

### PR DESCRIPTION
**Pull request by matejcik**
fixes #1409

this PR also updates `poetry.lock` so a bunch of new versions of things are used

only one test broke after _simply removing_ all `skip_ui` markers, and that was easy to fix by mocking out `os.urandom`, similar to how some tests already do

### CODE REVIEW

File: core/src/trezor/loop.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def _step(task: Task, value: Any) -> None:
     this_task = task
     try:
         if isinstance(value, BaseException):
-            result = task.throw(value)  # type: ignore
+            result = task.throw(value)
             # error: Argument 1 to "throw" of "Coroutine" has incompatible type "Exception"; expected "Type[BaseException]"
             # rationale: In micropython, generator.throw() accepts the exception object directly.
```
Comment (tsusanka) [inline core/src/trezor/loop.py:189]: If we remove `type: ignore` should we remove also this comment? Because it is explaining the ignore isn't it?

**Review by tsusanka — APPROVED** [review]
LGTM, one question and ship it :rocket:

File: core/src/trezor/loop.py (hunk 1 above)
Comment (matejcik) [reply core/src/trezor/loop.py:189]: that's a good point, and I'm actually wondering _why_ the error went away

Comment (prusnak) [reply core/src/trezor/loop.py:189]: maybe new version of mypy added implicit conversion from Exception to BaseException ?

Comment (matejcik) [reply core/src/trezor/loop.py:189]: but the problem was that we're passing an _instance_ where `throw()` expects a _type_

i'll look into it

Comment (matejcik) [reply core/src/trezor/loop.py:189]:
```
Revealed type is 'Overload(
def (Type[builtins.BaseException], builtins.object =, Union[types.TracebackType, None] =) -> Any,
def (builtins.BaseException, None =, Union[types.TracebackType, None] =) -> Any
)'
```
:woman_shrugging: i'll take it i guess. maybe a change in Python's definition of `throw()`?

Comment (matejcik) [reply core/src/trezor/loop.py:189]: welp! https://github.com/python/typeshed/issues/4241

Comment (matejcik) [reply core/src/trezor/loop.py]: 427b346

Comment (tsusanka) [reply core/src/trezor/loop.py]: Nice :)

**Review by prusnak — APPROVED** [review]
ACK when the tests are green and fixups are squashed

**Pull request merged**